### PR TITLE
docs: add quickstart for Pub/Sub emulator

### PR DESCRIPTION
**Pull request by anguillanneuf**
https://github.com/GoogleCloudPlatform/functions-framework-python/issues/23#issuecomment-833140050

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,31 @@ def function(request):
...
+   ```sh
+    export PUBSUB_PROJECT_ID=abc
+    gcloud beta emulators pubsub start \
+        --project=$PUBSUB_PROJECT_ID \
+        --host-port=localhost:8080
+   ```
+1. Create a Pub/Sub topic and attach a push subscription to the topic, using `http://localhost:8082` as its push endpoint. Publish some messages to the topic. Observe your function getting triggered by Pub/Sub messages.
+   ```sh
+    export TOPIC_ID_=may
+    export PUSH_SUBSCRIPTION_ID=five
```
Comment (grant) [inline README.md:147]: These values of `may` and `five` seem odd. Can we use more realistic variables?

File: README.md [hunk 2]
```diff
@@ -128,6 +128,31 @@ def function(request):
...
+    $(gcloud beta emulators pubsub env-init)
+    git clone https://github.com/googleapis/python-pubsub.git
+    cd python-pubsub/samples/snippets/
+    python publisher.py $PUBSUB_PROJECT_ID create $TOPIC_ID
+    python subscriber.py $PUBSUB_PROJECT_ID create $TOPIC_ID $PUSH_SUBSCRIPTION_ID http://localhost:8082
+    python publisher.py $PUBSUB_PROJECT_ID publish $TOPIC_ID
```
Comment (grant) [inline README.md:153]: Cloning the python sample repo and using a sample there from the Pub/Sub samples seems out of scope for a quickstart. Can we have a pure `python`, `gcloud` or `curl` test for this function?

File: README.md [hunk 3]
```diff
@@ -128,6 +128,31 @@ def function(request):
...
+    functions-framework --target=hello --debug --port=8082
+   ```
+1. Start the Pub/Sub emulator.
+   ```sh
+    export PUBSUB_PROJECT_ID=abc
+    gcloud beta emulators pubsub start \
+        --project=$PUBSUB_PROJECT_ID \
+        --host-port=localhost:8080
+   ```
+1. Create a Pub/Sub topic and attach a push subscription to the topic, using `http://localhost:8082` as its push endpoint. Publish some messages to the topic. Observe your function getting triggered by Pub/Sub messages.
```
Comment (grant) [inline README.md:144]: This step seems very large. Can we split it up into multiple steps?

File: README.md [hunk 4]
```diff
@@ -128,6 +128,31 @@ def function(request):
...
+    export TOPIC_ID_=may
+    export PUSH_SUBSCRIPTION_ID=five
+    $(gcloud beta emulators pubsub env-init)
+    git clone https://github.com/googleapis/python-pubsub.git
+    cd python-pubsub/samples/snippets/
+    python publisher.py $PUBSUB_PROJECT_ID create $TOPIC_ID
+    python subscriber.py $PUBSUB_PROJECT_ID create $TOPIC_ID $PUSH_SUBSCRIPTION_ID http://localhost:8082
+    python publisher.py $PUBSUB_PROJECT_ID publish $TOPIC_ID
+   ```
+
```
Comment (grant) [inline README.md:155]: What is the expected output of the program? Can we mention that here?

**Review by grant — CHANGES_REQUESTED** [review]
Thanks for the addition. I think there are some modifications we can make to improve the instructions for setup.

File: README.md [hunk 5]
```diff
@@ -128,6 +128,31 @@ def function(request):
...
 
+### Quickstart: With Pub/Sub emulator
+
+1. Start Functions Framework in your local development server listening on port 8082:
+   ```sh
+    functions-framework --target=hello --debug --port=8082
+   ```
+1. Start the Pub/Sub emulator.
+   ```sh
+    export PUBSUB_PROJECT_ID=abc
```
[the remark this answers is not in the document]
Comment (anguillanneuf) [reply README.md]: Updated to `my-project`. Many folks think they need to use their real GCP project ID but any alphanumeric strings would work.

File: README.md (hunk 2 above)
Comment (anguillanneuf) [reply README.md]: `gcloud` doesn't work with the Pub/Sub emulator.
`curl` would require folks to construct the message body. 
We recommend folks use our client libraries when interacting with the service. So the Python quickstart script is the best.

File: README.md (hunk 3 above)
Comment (anguillanneuf) [reply README.md]: I put these steps together because there are multiple terminals used in this quickstart. All these steps should be run in the same terminal. And they all do Pub/Sub-related things. I vote we keep them together.

**Review by grant — APPROVED** [review]
Thanks, looks alright to me. I didn't test it, but I assume it works as expected when copying and pasting.

File: README.md [hunk 6]
```diff
@@ -128,6 +128,68 @@ def function(request):
 This function will catch the `ZeroDivisionError` and return a different
 response instead.
 
+### Quickstart: Pub/Sub emulator
```
Comment (grant) [inline README.md:131]: Can we have uniform newlines?
---

### Quickstart: Pub/Sub emulator

1. Create a `main.py` file with the following contents:

   ```python
   def hello(request):
        return "Hello world!"
   ```

1. Start the Functions Framework on port 8080:

   ```sh
   functions-framework --target=hello --debug --port=8080
   ```

1. Start the Pub/Sub emulator on port 8085.

   ```sh
   export PUBSUB_PROJECT_ID=my-project
   gcloud beta emulators pubsub start \
       --project=$PUBSUB_PROJECT_ID \
       --host-port=localhost:8085
   ```
   You should see the following after the Pub/Sub emulator has started successfully:
   ```none
   [pubsub] INFO: Server started, listening on 8085
   ```

1. Create a Pub/Sub topic and attach a push subscription to the topic, using `http://localhost:8085` as its push endpoint. [Publish](https://cloud.google.com/pubsub/docs/quickstart-client-libraries#publish_messages) some messages to the topic. Observe your function getting triggered by the Pub/Sub messages.

   ```sh
   export TOPIC_ID=my-topic
   export PUSH_SUBSCRIPTION_ID=my-subscription
   $(gcloud beta emulators pubsub env-init)
   git clone https://github.com/googleapis/python-pubsub.git
   cd python-pubsub/samples/snippets/
    
   python publisher.py $PUBSUB_PROJECT_ID create $TOPIC_ID   
   python subscriber.py $PUBSUB_PROJECT_ID create-push $TOPIC_ID $PUSH_SUBSCRIPTION_ID http://localhost:8085
   python publisher.py $PUBSUB_PROJECT_ID publish $TOPIC_ID
   ```
   You should see the following after the commands have run successfully:
   ```none
   Created topic: projects/my-project/topics/my-topic
   
   topic: "projects/my-project/topics/my-topic"
   push_config {
     push_endpoint: "http://localhost:8085"
   }
   ack_deadline_seconds: 10
   message_retention_duration {
     seconds: 604800
   }
   .
   Endpoint for subscription is: http://localhost:8085
    
   1
   2
   3
   4
   5
   6
   7
   8
   9
   Published messages to projects/my-project/topics/my-topic.
   ```
